app/cdap/components/NamespaceAdmin/Connections.tsx

```diff
@@ -80,14 +88,79 @@ const ConnectionsView: React.FC<IConnectionsProps> = ({ connections }) => {
     setDeleteError(null);
   }
 
+  function toggleConnectionCreate() {
+    const newState = !isCreateConnectionOpen;
+
+    setIsCreateConnectionOpen(newState);
+
+    if (!newState) {
+      setInitConnConfig(null);
+    }
+  }
+
+  function getConnectionConfig(conn) {
+    const connectionConfig = {
+      name: conn.name,
+      description: conn.description,
+      category: conn.plugin.category,
+      plugin: conn.plugin,
+    };
+
+    return connectionConfig;
+  }
+
+  function exportConnection(conn) {
+    const connectionConfig = getConnectionConfig(conn);
+    const artifact = objectQuery(conn, 'plugin', 'artifact') || {};
+    DownloadFile(
+      connectionConfig,
+      null,
+      `${conn.name}-connector-${artifact.name}-${artifact.version}`
+    );
+  }
+
+  function cloneConnection(conn) {
+    const config = getConnectionConfig(conn);
+    delete config.name;
+
+    openConnectionConfig(config);
+  }
+
+  function handleFile(event) {
+    if (!objectQuery(event, 'target', 'files', 0)) {
+      return;
+    }
+    const uploadedFile = event.target.files[0];
+
+    const reader = new FileReader();
+    reader.readAsText(uploadedFile, 'UTF-8');
+
+    reader.onload = (evt) => {
+      try {
+        const config = JSON.parse(evt.target.result.toString());
+        openConnectionConfig(config);
+      } catch (e) {
+        // tslint:disable-next-line: no-console
+        console.log('Error parsing imported file', e);
```

Comment:
Can we add an error `Alert` here? It is these flows that will be hard for users to debug. Understand we don't have the error handling hashed out, don't want to miss out on this while we are adding the functionality on a fairly simple page.